kolibri/content/models.py

```diff
@@ -0,0 +1,330 @@
+                pass
+            self.checksum = None
+            self.available = False
+            self.file_size = None
+            self.extension = None
+        super(File, self).save(*args, **kwargs)
+
+class License(AbstractContent):
+    """
+    Normalize the license of ContentMetadata model
+    """
+    license_name = models.CharField(max_length=50)
+
+    class Admin:
+        pass
+
+    def __str__(self):
+        return self.license_name
+
+class ContentRelationship(AbstractContent):
+    """
+    Provide a abstract model for defining any relationships between two ContentMetadata objects.
+    """
+    contentmetadata_1 = models.ForeignKey(ContentMetadata, related_name='%(app_label)s_%(class)s_1')
+    contentmetadata_2 = models.ForeignKey(ContentMetadata, related_name='%(app_label)s_%(class)s_2')
+
+    class Meta:
+        abstract = True
+
+    class Admin:
+        pass
+
+class PrerequisiteContentRelationship(ContentRelationship):
+    """
+    Predefine the prerequisite relationship between two ContentMetadata objects.
+    """
+    class Meta:
+        unique_together = ['contentmetadata_1', 'contentmetadata_2']
+
+    class Admin:
+        pass
+
+    def clean(self, *args, **kwargs):
+        # self reference exception
+        if self.contentmetadata_1 == self.contentmetadata_2:
+            raise IntegrityError('Cannot self reference as prerequisite.')
+        # immediate cyclic exception
+        elif PrerequisiteContentRelationship.objects.using(self._state.db)\
+                .filter(contentmetadata_1=self.contentmetadata_2, contentmetadata_2=self.contentmetadata_1):
+            raise IntegrityError(
+                'Note: Prerequisite relationship is directional! %s and %s cannot be prerequisite of each other!'
+                % (self.contentmetadata_1, self.contentmetadata_2))
+        # distant cyclic exception
+        # elif <this is a nice to have exception, may implement in the future when the priority raises.>
+        #     raise Exception('Note: Prerequisite relationship is acyclic! %s and %s forms a closed loop!' % (self.contentmetadata_1, self.contentmetadata_2))
+        super(PrerequisiteContentRelationship, self).clean(*args, **kwargs)
+
+    def save(self, *args, **kwargs):
+        self.full_clean()
+        super(PrerequisiteContentRelationship, self).save(*args, **kwargs)
+
+
+class RelatedContentRelationship(ContentRelationship):
+    """
+    Predefine the related relationship between two ContentMetadata objects.
+    """
+    class Meta:
+        unique_together = ['contentmetadata_1', 'contentmetadata_2']
+
+    class Admin:
+        pass
+
+    def clean(self, *args, **kwargs):
+        # self reference exception
+        if self.contentmetadata_1 == self.contentmetadata_2:
+            raise IntegrityError('Cannot self reference as related.')
+        # immediate cyclic exception
+        elif RelatedContentRelationship.objects.using(self._state.db)\
```

Comment:
Related content is not unidirectional, but rather bidirectional, so this is not actually a problem here.

The issue is that we can't enforce the relationship to be symmetric in Django - so we are probably better off allowing these to be set, and proxying the 'related' property to be a unique combination of the related, and reverse_related properties for that object.

Opened an issue here for this: #32